lib/functions.sh

```diff
@@ -236,6 +236,22 @@ function get_field {
     done
 }
 
+# Usage: get_or_create_domain <name> <description>
+function get_or_create_domain {
+    local domain_id
+    # Gets domain id
+    domain_id=$(
+        # Gets domain id
+        openstack domain show $1 \
+            -f value -c id 2>/dev/null ||
+        # Creates new domain
+        openstack domain create $1 \
+            --description "$2" \
+            -f value -c id
+    )
```

Comment:
I think this can be replaced with a single line:
domain_id=$(`openstack domain create $1 --description "$2' --or-show -f value -c id`)